src/core/ddsi/src/ddsi_plist.c

```diff
@@ -374,11 +374,18 @@ static dds_return_t deser_locator (void * __restrict dst, size_t * __restrict ds
   struct dd tmpdd = *dd;
   tmpdd.buf += *srcoff;
   tmpdd.bufsz -= *srcoff;
-  if (do_locator (x, *flagset->present, flagset->wanted, flag, &tmpdd, dd->factory) < 0)
-    return DDS_RETCODE_BAD_PARAMETER;
+  switch (do_locator (x, *flagset->present, flagset->wanted, flag, &tmpdd, dd->factory))
+  {
+    case DOLOC_INVALID:
+      return DDS_RETCODE_BAD_PARAMETER;
+    case DOLOC_IGNORED:
+      break;
+    case DOLOC_ACCEPTED:
+      *flagset->present |= flag;
+      break;
+  }
   *srcoff += 24;
   *dstoff += sizeof (*x);
```

Comment:
Should the value of `dstoff` be increased in case the locator is ignored, because the locator will not be in `dst`? (it's not really a problem, because `dstoff` is currently not used after this function).

Reply:
Looks to me like this is the case for both `srcoff` and `dstoff`. With the value never used after the call, I'd say those really should be pass-by-value. I'll merge this PR as-is, and do a separate PR to change the types, update all (few) specialised deserialisers and update the call site.